components/service-operator/apis/storage/v1beta1/image_repository_types.go

```diff
@@ -0,0 +1,149 @@
+	Items           []ImageRepository `json:"items"`
+}
+
+// +kubebuilder:object:root=true
+
+// ImageRepository is the Schema for the ImageRepository API
+type ImageRepository struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec          ImageRepositorySpec `json:"spec,omitempty"`
+	object.Status `json:"status,omitempty"`
+}
+
+// Name returns the name of the ImageRepository cloudformation stack
+func (s *ImageRepository) GetStackName() string {
+	return fmt.Sprintf("%s-%s-%s-%s", env.ClusterName(), "ecr", s.Namespace, s.ObjectMeta.Name)
+}
+
+// SecretName returns the name of the secret that will be populated with data
+func (s *ImageRepository) GetSecretName() string {
+	if s.Spec.Secret == "" {
+		return s.GetName()
+	}
+	return s.Spec.Secret
+}
+
+// Template returns a cloudformation Template for provisioning an ImageRepository
+func (s *ImageRepository) GetStackTemplate() *cloudformation.Template {
+	template := cloudformation.NewTemplate()
+
+	template.Parameters[IAMRoleParameterName] = map[string]string{
+		"Type": "String",
+	}
+
+	repositoryName := fmt.Sprintf("%s-%s-%s", env.ClusterName(), s.Namespace, s.ObjectMeta.Name)
+	template.Resources[ImageRepositoryResourceName] = &cloudformation.AWSECRRepository{
+		RepositoryName: repositoryName,
+	}
+
+	imageRepositoryArn := cloudformation.Join("", []string{cloudformation.GetAtt(ImageRepositoryResourceName, "Arn")})
+
+	template.Resources[ImageRepositoryResourceIAMPolicy] = &cloudformation.AWSIAMPolicy{
+		PolicyName:     cloudformation.Join("-", []string{"ecr", repositoryName}),
+		PolicyDocument: cloudformation.NewRolePolicyDocument([]string{imageRepositoryArn}, []string{"ecr:*"}),
+		Roles: []string{
+			cloudformation.Ref(IAMRoleParameterName),
+		},
+	}
+
+	template.Outputs[ImageRepositoryName] = map[string]interface{}{
+		"Description": "Image repository name to be returned to the user.",
+		"Value":       cloudformation.Ref(ImageRepositoryResourceName),
+	}
+
+	template.Outputs[ImageRepositoryURI] = map[string]interface{}{
+		"Description": "Image repository URI to be returned to the user.",
+		"Value":       cloudformation.Join("", []string{cloudformation.Ref(AccountIdParameterName), ".dkr.ecr.eu-west-2.amazonaws.com/", repositoryName}),
+	}
+
+	template.Outputs[ImageRepositoryRegion] = map[string]interface{}{
+		"Description": "Region that the image repository lives in",
+		"Value":       "eu-west-2",
+	}
+
+	template.Outputs[IAMRoleParameterName] = map[string]interface{}{
+		"Description": "Name of the IAM role with access to the image repository",
+		"Value":       cloudformation.Ref(IAMRoleParameterName),
+	}
+
+	template.Outputs[IAMRoleArnParameterName] = map[string]interface{}{
+		"Description": "ARN of the IAM role with access to the image repository",
+		"Value":       cloudformation.Join("", []string{"arn:aws:iam::", cloudformation.Ref(AccountIdParameterName), ":role/", cloudformation.Ref(IAMRoleParameterName)}),
```

Comment:
`cloudformation.GetAtt(IAMRoleParameterName, "Arn")` ?

Reply:
input parameters do not have attributes.